src/ghost_borrow_mut.rs

```diff
@@ -134,18 +134,18 @@ impl<'a, 'brand, T, const N: usize> GhostBorrowMut<'a, 'brand> for &'a [GhostCel
     }
 }
 
-impl<'a, 'brand, T: ?Sized, const N: usize> GhostBorrowMut<'a, 'brand> for [&'a GhostCell<'brand, T>; N] {
+impl<'a, 'brand, T: sealed::GetSpan + ?Sized, const N: usize> GhostBorrowMut<'a, 'brand> for [&'a GhostCell<'brand, T>; N] {
     type Result = [&'a mut T; N];
     type Error = GhostAliasingError;
 
     fn borrow_mut(self, token: &'a mut GhostToken<'brand>) -> Result<Self::Result, Self::Error> {
-        //  Safety:
-        //  -   `[&'a GhostCell<'brand, T>; N]` and `[*const T; N]` have the same size.
-        //  -   `[&'a GhostCell<'brand, T>; N]` implements `Copy`, so no `mem::forget` is needed.
-        //  -   We can't use `mem::transmute`, because of https://github.com/rust-lang/rust/issues/61956.
-        //  Note: We can't cast to a `*const [*const (); N]` to immediately extract addresses due to the size/layout of fat pointers;
-        //        see https://github.com/matthieu-m/ghost-cell/pull/16#discussion_r884114694 for more details.
-        check_distinct(unsafe { ptr::read(&self as *const _ as *const [*const T; N]) })?;
+        let mut spans = [sealed::Span::Single(ptr::null()); N];
```

Comment:
Why not use the [map function](https://doc.rust-lang.org/std/primitive.array.html#method.map)?

Reply:
Ah! I keep forgetting about those const generics enabling more on arrays.